src/org/opendatakit/briefcase/export/CsvMapper.java

```diff
@@ -0,0 +1,352 @@
+   * with some data types, like the {@link DataType#GEOPOINT}, which need to be represented with
+   * 4 columns on a CSV file.
+   *
+   * @param fqn        the {@link String} FQN of the CSV column
+   * @param outputSize the wanted {@link Integer} output size
+   * @return a {@link List} of {@link Pair} instances to represent that some CSV values will be empty.
+   */
+  static List<Pair<String, String>> empty(String fqn, int outputSize) {
+    return IntStream.range(0, outputSize).boxed().map(__ -> Pair.of(fqn, (String) null)).collect(toList());
+  }
+
+  /**
+   * Produces a {@link List} of CSV compatible {@link String} lines with the values of
+   * a {@link Submission} instance's particular repeat group.
+   *
+   * @param groupModel      the repeat group's {@link Model}
+   * @param elements        a {@link List} of {@link XmlElement} instances to be transformed into a CSV line
+   * @param exportMedia     a {@link Boolean} indicating if we have to export media files too
+   * @param exportMediaPath the {@link Path} to the output media dir
+   * @param instanceId      the instance ID of the {@link Submission} instance that it's being exported
+   * @param workingDir      the {@link Path} to the working directory of the {@link Submission} instance that it's being exported
+   * @return a {@link List} of CSV compatible {@link String} lines
+   */
+  static String getRepeatCsvLine(Model groupModel, List<XmlElement> elements, boolean exportMedia, Path exportMediaPath, String instanceId, Path workingDir) {
+    StringBuilder sb = new StringBuilder();
+    elements.forEach(element -> {
+      String localId = element.getCurrentLocalId(instanceId);
+      StringBuilder sb2 = new StringBuilder();
+      groupModel.forEach(field -> getMapper(field).apply(
+          localId,
+          workingDir,
+          field,
+          element.findElement(field.getName()),
+          exportMedia,
+          exportMediaPath
+      ).forEach(pair -> sb2.append(",").append(encode(pair.right, pair.left.startsWith("meta") || pair.left.startsWith("SET-OF")))));
+
+      sb2.append(",").append(encode(element.getParentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getCurrentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getGroupLocalId(instanceId), false));
+      sb.append(sb2.toString().substring(1)).append("\n");
+    });
+    return sb.toString();
+  }
+
+  /**
+   * Tells how nulls should be encoded depending on the {@link DataType}.
+   * <p>
+   * Currently, only geopoints and date/time fields allow nulls as empty
+   * cols. The rest require an empty string.
+   */
+  private static boolean nullsAsEmptyCols(DataType dataType) {
+    switch (dataType) {
+      case GEOPOINT:
+      case DATE:
+      case TIME:
+      case DATE_TIME:
+        return true;
+      default:
+        return false;
+    }
+  }
```

Comment:
I find this much simpler. You?

![screen shot 2018-05-17 at 2 17 58 pm](https://user-images.githubusercontent.com/55099/40204388-321cd844-59dd-11e8-9204-266a63f651f8.png)